bigtable/client/metadata_update_policy_test.cc

```diff
@@ -13,23 +13,81 @@
 // limitations under the License.
 
 #include "bigtable/client/metadata_update_policy.h"
+#include "bigtable/client/admin_client.h"
+#include "bigtable/client/internal/make_unique.h"
+#include "bigtable/client/table.h"
+#include "bigtable/client/table_admin.h"
+#include "bigtable/client/testing/embedded_server_test_fixture.h"
 #include <gtest/gtest.h>
+#include <thread>
 
-std::string const kInstanceName = "projects/foo-project/instances/bar-instance";
-std::string const kTableId = "baz-table";
-std::string const kTableName =
-    "projects/foo-project/instances/bar-instance/tables/baz-table";
+namespace btproto = google::bigtable::v2;
+namespace adminproto = google::bigtable::admin::v2;
+
+namespace {
+class MetadataUpdatePolicyTest
+    : public bigtable::testing::EmbeddedServerTestFixture {};
+}  // anonymous namespace
+
+/// @test A test for setting metadata for admin operations.
+TEST_F(MetadataUpdatePolicyTest, RunWithEmbeddedServer) {
+  grpc::string expected = "parent=" + kInstanceName;
+  auto gc = bigtable::GcRule::MaxNumVersions(42);
+  admin_->CreateTable(kTableName, bigtable::TableConfig({{"fam", gc}}, {}));
+  // Get metadata from embedded server
+  auto client_metadata = admin_service_.client_metadata();
+  auto range = client_metadata.equal_range("x-goog-request-params");
+  // first check to see if client_metadata has only one occurance of parameter.
+  EXPECT_EQ(1, std::distance(range.first, range.second));
+  for (auto it = range.first; it != range.second; ++it) {
```

Comment:
This can be simplified to:

```C++
ASSERT_EQ(1, std::distance(range.first, range.second);
std::string actual = range.first->second;
EXPECT_EQ(expected, range.first->second);
```

After all you just checked that there is only one element in the iteration.

Reply:
changed